pkg/instance/service.go

```diff
@@ -184,8 +186,60 @@ func (s service) unlink(id uint) error {
 	return s.instanceRepository.Unlink(instance)
 }
 
+func (s service) matchRequiredParameters(stackParams []model.StackRequiredParameter, instanceParams []model.InstanceRequiredParameter) []string {
+	unmatchedParams := make([]string, 0)
+	paramNames := make(map[string]struct{})
+
+	for _, param := range stackParams {
+		paramNames[param.Name] = struct{}{}
+	}
+
+	for _, param := range instanceParams {
+		if _, ok := paramNames[param.StackRequiredParameterID]; !ok {
+			unmatchedParams = append(unmatchedParams, param.StackRequiredParameterID)
+		}
+	}
+
+	return unmatchedParams
+}
+
+func (s service) matchOptionalParameters(stackParams []model.StackOptionalParameter, instanceParams []model.InstanceOptionalParameter) []string {
+	unmatchedParams := make([]string, 0)
+	paramNames := make(map[string]struct{})
+
+	for _, param := range stackParams {
+		paramNames[param.Name] = struct{}{}
+	}
+
+	for _, param := range instanceParams {
+		if _, ok := paramNames[param.StackOptionalParameterID]; !ok {
+			unmatchedParams = append(unmatchedParams, param.StackOptionalParameterID)
+		}
+	}
+
+	return unmatchedParams
+}
+
 func (s service) Save(instance *model.Instance) (*model.Instance, error) {
-	err := s.instanceRepository.Save(instance)
+	instanceStack, err := s.stackService.Find(instance.StackName)
+	if err != nil {
+		return nil, err
+	}
+
+	unmatchedReqParams := s.matchRequiredParameters(instanceStack.RequiredParameters, instance.RequiredParameters)
+	unmatchedOptParams := s.matchOptionalParameters(instanceStack.OptionalParameters, instance.OptionalParameters)
+
+	unmatchedParams := make([]string, 0)
+	unmatchedParams = append(unmatchedParams, unmatchedReqParams...)
+	unmatchedParams = append(unmatchedParams, unmatchedOptParams...)
+
+	if len(unmatchedParams) > 0 {
+		return nil, apperror.NewBadRequest(
+			fmt.Sprintf("these parameters were not found for the \"%s\" stack: %v", instance.StackName, unmatchedParams),
```

Comment:
```suggestion
			fmt.Sprintf("parameters %q are not valid parameters for stack %q", instance.StackName, unmatchedParams),
```

`%q` quotes for us 😄 https://pkg.go.dev/fmt#hdr-Printing